src/MysqlConfig.php

```diff
@@ -157,24 +161,40 @@ public function withConnectContext(ConnectContext $context): self
         return $new;
     }
 
-    public function getCharset(): string
+    public function getCharset(): ?string
```

Comment:
This is a breaking change. Shouldn't we keep a client default?

Reply:
Yes this is a breaking change.
Keeping a default charset make no sens. The charset change depending on the installation of MySQL.

MySQL set by default utf8mb4 and utf8mb4_0900_ai_ci as default [charset since MySQL 8.0](https://dev.mysql.com/doc/refman/8.0/en/charset-server.html), but I never see this configuration on installations.

Reply:
That's actually a sane default.